sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp

```diff
@@ -519,6 +520,9 @@ ur_result_t urDeviceGetInfo(
     return ReturnValue(Device->Platform->ZeDriverVersion.c_str());
   case UR_DEVICE_INFO_VERSION:
     return ReturnValue(Device->Platform->ZeDriverApiVersion.c_str());
+  case UR_DEVICE_INFO_BACKEND_RUNTIME_VERSION:
+    return ReturnValue(
+        std::to_string(SYCL_EXT_ONEAPI_BACKEND_LEVEL_ZERO).c_str());

```

Comment:
if this information is an intrinsic knowledge of SYCL then this query should be handled there (and not in plugins)

Reply:
@bmyates was looking into this while working on the UR CTSs. We were thinking on returning here the driver version of the L0 driver.

Reply:
SYCL says:
> Returns a string describing the version of the [SYCL backend](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#backend) associated with the [device](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#device). The possible values are specified in the [SYCL backend](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#backend) specification of the [SYCL backend](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#backend) associated with the [device](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#device).

And we have SYCL L0 backend spec here https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_backend_level_zero.md

And that has a version (on 5 currently), which I think is what this wants us to return.
If that's the case, then we don't need to go to PI at all, and just return what SYCL knows/implements by itself.

Reply:
If this property is not intrinsic to the UR adapter, should UR_DEVICE_INFO_BACKEND_RUNTIME_VERSION be removed from UR specification?

Reply:
> If this property is not intrinsic to the UR adapter, should UR_DEVICE_INFO_BACKEND_RUNTIME_VERSION be removed from UR specification?

I think so, but let's confirm what we want to return for `info::device::backend_version`.
@gmlueck : can you please take a look?

Reply:
Returning the version of the `SYCL_EXT_ONEAPI_BACKEND_LEVEL_ZERO` feature-test macro does not seem that useful because you can get the same information from the macro.

Offhand, I can't think of anything useful to return from `info::device::backend_version`.  I think we could just document in our backend interop spec that we don't return anything for this query.  Our implementation should just return an empty string in that case.

If we decide later to return something, I presume we can add a matching UR API at that time.

Reply:
Thanks @gmlueck! I made some addition to the spec, could you please review? Most likely, some more words are required for this.